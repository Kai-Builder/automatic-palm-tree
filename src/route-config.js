import {ImplicitCallback, Security} from '@okta/okta-react';
import {Route, BrowserRouter as Router} from 'react-router-dom';
import {PageContainer} from './page-container';
import React from 'react';

const oktaConfig = {
  clientId: '0oaf3emz9n7G0xYA20h7',
  issuer: 'https://dev-689082.oktapreview.com/oauth2/default',
  path: '/implicit/callback',
  redirectUri: `${window.location.origin}/implicit/callback`
};

export const RouteConfig = () => (
  <Router>
    <Security
      client_id={oktaConfig.clientId}
      issuer={oktaConfig.issuer}
      redirect_uri={oktaConfig.redirectUri}
    >
      <PageContainer/>
      <Route
        component={ImplicitCallback}
        path={oktaConfig.path}
      />
    </Security>
  </Router>
);
