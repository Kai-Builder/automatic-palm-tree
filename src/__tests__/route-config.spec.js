import React from 'react';
import {RouteConfig} from '../route-config';
import {shallow} from 'enzyme';

describe('RouteConfig', () => {
  const requiredProps = () => ({
  });

  const render = (props = requiredProps()) => shallow(<RouteConfig {...props}/>);

  test('component renders', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });
});
