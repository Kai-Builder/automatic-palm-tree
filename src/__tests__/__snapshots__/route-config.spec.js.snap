// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteConfig component renders 1`] = `
<BrowserRouter>
  <withRouter(Security)
    client_id="0oaf3emz9n7G0xYA20h7"
    issuer="https://dev-689082.oktapreview.com/oauth2/default"
    redirect_uri="http://localhost/implicit/callback"
  >
    <PageContainer />
    <Route
      component={[Function]}
      path="/implicit/callback"
    />
  </withRouter(Security)>
</BrowserRouter>
`;
